### ISSUE TITLE
Track Type database migrations not applied

### ISSUE THREAD

**Issue by itcrowdsource** [LibreTime/libretime#990]
**Describe the bug**
After clicking the login button this message appears:

Oops!
Something went wrong!

**To Reproduce**
1. Install
2. Login

**Expected behavior**
Be able to login

**LibreTime version:**
Latest version from GIT

**Installation method:**
 - OS: Ubuntu 
 - OS Version 18.04 Bionic
 - Method: ./install`

**Screenshots**
![image](https://user-images.githubusercontent.com/20809792/77837220-888c9300-715e-11ea-8f72-64ef7101b874.png)

**Additional context**
Zendphp.log

```
2020-03-28T22:38:33+00:00 ERR (3): <url> [ErrorController.php:54 - errorAction()] - An internal application error has occurred.: PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR: relation “cc_track_types” does not exist
LINE 1: SELECT id, code, type_name, description FROM cc_track_types …
^ in /usr/share/airtime/php/airtime_mvc/application/common/Database.php:24
Stack trace:
#0 /usr/share/airtime/php/airtime_mvc/application/common/Database.php(24): PDOStatement->execute()
#1 /usr/share/airtime/php/airtime_mvc/application/models/Tracktype.php(124): Application_Common_Database::prepareAndExecute(‘SELECT id, code…’, Array, ‘all’)
#2 /usr/share/airtime/php/airtime_mvc/application/models/Tracktype.php(98): Application_Model_Tracktype::getTracktypesData(Array, NULL)
#3 /usr/share/airtime/php/airtime_mvc/application/controllers/plugins/PageLayoutInitPlugin.php(146): Application_Model_Tracktype::getTracktypes()
#4 /usr/share/airtime/php/airtime_mvc/application/controllers/plugins/PageLayoutInitPlugin.php(85): PageLayoutInitPlugin->_initTranslationGlobals(Object(Zend_View))
#5 /usr/share/airtime/php/airtime_mvc/application/controllers/plugins/PageLayoutInitPlugin.php(47): PageLayoutInitPlugin->_initGlobals()
#6 /usr/share/airtime/php/vendor/zf1s/zend-controller/library/Zend/Controller/Plugin/Broker.php(260): PageLayoutInitPlugin->routeShutdown(Object(Zend_Controller_Request_Http))
#7 /usr/share/airtime/php/vendor/zf1s/zend-controller/library/Zend/Controller/Front.php(923): Zend_Controller_Plugin_Broker->routeShutdown(Object(Zend_Controller_Request_Http))
#8 /usr/share/airtime/php/vendor/zf1s/zend-application/library/Zend/Application/Bootstrap/Bootstrap.php(105): Zend_Controller_Front->dispatch()
#9 /usr/share/airtime/php/vendor/zf1s/zend-application/library/Zend/Application.php(384): Zend_Application_Bootstrap_Bootstrap->run()
#10 /usr/share/airtime/php/airtime_mvc/application/airtime-boot.php(84): Zend_Application->run()
#11 /usr/share/airtime/php/airtime_mvc/public/index.php(68): require_once(’/usr/share/airt…’)
```

**Comment by paddatrapper** [comment]
Seems like the database migrations required by #892 haven't been applied here...

**Comment by itcrowdsource** [comment]
Problem solved by creating a database table:

```
CREATE TABLE IF NOT EXISTS cc_track_types
(
id integer NOT NULL,
code VARCHAR(16) NOT NULL,
type_name VARCHAR(64),
description VARCHAR(255),
visibility boolean DEFAULT true NOT NULL,
CONSTRAINT cc_track_types_pkey PRIMARY KEY (id),
CONSTRAINT cc_track_types_code_key UNIQUE (code)
);

INSERT INTO cc_track_types VALUES (1, ‘MUS’, ‘Music’, ‘This is used for tracks containing music.’, true);
INSERT INTO cc_track_types VALUES (2, ‘SID’, ‘Station ID’, ‘This is used for Station IDs’, true);
INSERT INTO cc_track_types VALUES (3, ‘INT’, ‘Show Intro’, ‘This can be used for organizing all the show introductions.’, true);
INSERT INTO cc_track_types VALUES (4, ‘OUT’, ‘Show Outro’, ‘This can be used for organizing all the show outroductions.’, true);
INSERT INTO cc_track_types VALUES (5, ‘SWP’, ‘Sweeper’, ‘This is used for segues between songs.’, true);
INSERT INTO cc_track_types VALUES (6, ‘JIN’, ‘Jingle’, ‘A short song or tune, normally played during commercial breaks. Contains one or more hooks.’, true);
INSERT INTO cc_track_types VALUES (7, ‘PRO’, ‘Promo’, ‘For promotional use.’, true);
INSERT INTO cc_track_types VALUES (8, ‘SHO’, ‘Shout Out’, 'A message of congratulation, greeting. support, or appreciation. ', true);
INSERT INTO cc_track_types VALUES (9, ‘NWS’, ‘News’, ‘This is used for noteworthy information, announcements.’, true);
INSERT INTO cc_track_types VALUES (10, ‘COM’, ‘Commercial’, ‘This is used for commerical advertising.’, true);
INSERT INTO cc_track_types VALUES (11, ‘ITV’, ‘Interview’, ‘This is used for radio interviews’, true);
INSERT INTO cc_track_types VALUES (12, ‘VTR’, ‘Voice Tracking’, ‘Also referred as robojock or taped. Make announcements without actually being in the station.’, true);
```

**Comment by paddatrapper** [comment]
This should be done automatically. We need to check why that isn't happening

**Comment by Robbt** [comment]
So I just tested this on a Ubuntu Xenial vagrant box & the first part happens
```ALTER TABLE cc_files ADD COLUMN track_type VARCHAR(16);
```
 but alas the cc_track_types table is not created.
So there is an issue here with how the SQL is being parsed by Zend1 / LibreTime.

**Comment by Robbt** [comment]
Ok so this is the error I get when trying to run the code contained in the update on an existing database.
```

airtime=> CREATE TABLE IF NOT EXISTS "cc_track_types"
airtime-> (
airtime(>     "id" integer DEFAULT nextval('cc_track_types_id_seq'::regclass) NOT NULL,
airtime(>     "code" VARCHAR(16) NOT NULL,
airtime(>     "type_name" VARCHAR(64),
airtime(>     "description" VARCHAR(255),
airtime(>     "visibility" boolean DEFAULT true NOT NULL,
airtime(>     CONSTRAINT "cc_track_types_pkey" PRIMARY KEY ("id"),
airtime(>     CONSTRAINT "cc_track_types_code_key" UNIQUE ("code")
airtime(> );
ERROR:  relation "cc_track_types_id_seq" does not exist

```
Looks like we are attempting to create an auto incrementing ID 
I'll see how it has been done for other tables being created but we might want to use SERIAL as the type based upon this https://www.tutorialspoint.com/postgresql/postgresql_using_autoincrement.htm

**Comment by Robbt** [comment]
Yeah the proper way to do this would be to use serial - https://github.com/LibreTime/libretime/blob/e584f859feb98412f9bff3e9b588e7079d987bb7/airtime_mvc/application/controllers/upgrade_sql/airtime_2.5.16/upgrade.sql#L33

So we have to decide whether we will just fix the existing code upgrade considering that whomever updated to the latest git probably posted on the forum and/or saw the fix and so it will work for anyone upgrading in the future.

That is what I'll do in a PR and test. Sorry to whomever got hit by this for not testing this properly before merging.

**Comment by paddatrapper** [comment]
I think that is probably best

**Comment by paddatrapper** [comment]
Fixed in #1011
